services/QuillLMS/app/models/subscription.rb

```diff
@@ -117,6 +117,8 @@ class RenewalNilStripeCustomer < StandardError; end
   validates :stripe_invoice_id, allow_blank: true, stripe_uid: { prefix: :in }
   validates :stripe_subscription_id, allow_blank: true, stripe_uid: { prefix: :sub }
 
+  before_validation :strip_stripe_id_whitespace, only: [:stripe_invoice_id, :strip_subscription_id]
```

Comment:
I think you're missing an e in `stripe_subscription_id` here. I can't find `only` in the context of validations in the Rails docs - I think it maybe isn't doing anything at all and you want something more like `if: -> { will_save_change_to_stripe_invoice_id? || will_save_change_to_stripe_subscription_id? }`

Reply:
Good catch.  I tried a couple of different options with `if` to restrict when this runs, but none of them were all that readable/functional.  I think the thing to do here is to just always run this hook since it's fast and safe to run whenever.